routing/pathfind.go

```diff
@@ -624,14 +624,11 @@ func findPath(g *graphParams, r *restrictParams,
 			return
 		}
 
-		// If the edge has no time lock delta, the payment will always
-		// fail, so return.
-		//
-		// TODO(joostjager): Is this really true? Can't it be that
-		// nodes take this risk in exchange for a extraordinary high
-		// fee?
+		// Every edge should have a positive time lock delta. If we
+		// encounter a zero delta, log a warning line.
```

Comment:
I don't know all causes of the delta being zero. Don't even know if it happens at all. But I think there should be no check here and not even a log line. We just need to trust the systems that put the value in place. If we start validating data everywhere we use it, it will be never ending.

My idea was to put the log line here for now and see if that warnings comes up.

Reply:
The spec does allow a zero value it seems. I guess there might even exists cases where it would make sense with a 0 timelock delta?

Anyway, I agree that this is a good solution for now :) (since intermediate nodes having a 0 delta shouldn't negatively affect us)